ocaml/testsuite/tests/typing-layouts-float32/basics_beta.ml

```diff
@@ -0,0 +1,907 @@
+(* TEST
+ flambda2;
+ {
+   flags = "-extension layouts_alpha -extension small_numbers";
+   expect;
+ }{
+   flags = "-extension layouts_beta -extension small_numbers";
+   expect;
+ }
+*)
+
+(* This test is almost entirely a copy of [basics.ml], except
+   with different output for some layouts-beta features.
+   You should diff this file against [basics.ml] to see what's
+   different.
+*)
```

Comment:
The diff between `basics.ml` and this file is larger than I expected: there are a number of tests at the bottom that are only in one. (And some comments have drifted.)

Reply:
Yeah, the bottom block of tests is present in `typing-layouts-float64/basics.ml`, but it needs to be in `typing-layouts-float32/basics_beta.ml` instead of basics because it requires small_numbers.

The comment diff is because I replicated basics_beta from float64, which had updated comments. 
I think the diff is correct, though it makes the advice to diff with basics a bit confusing...

Reply:
I think we should make the diff in comments go away in favor of what's in `basics_beta.ml` (it sounds like also in the `-float64` directory, which I don't think has the diff intentionally?)

Reply:
Now fixed